romdisk/date.asm

```diff
@@ -50,6 +88,75 @@ date_main:
         WRITE()
         ret
 
+        ; Read ACSII date and fill out date part of DATE_STRUCT.
+        ; The input format will be as follows:
+        ;   YYYY-MM-DD
+        ; Does not check the '-' chars, they can be anything. Does not
+        ; validate number ranges.
+        ; Parameters:
+        ;       HL - Pointer to the date structure, of size DATE_STRUCT_SIZE
+        ;       DE - String source.
+        ; Returns:
+        ;       The date parts of the input date struct have been filled out.
+        ;       DE - DE + 10
+        PUBLIC ascii_to_date
+ascii_to_date:
+        push hl
+        call read_bcd_pair
+        call read_bcd_pair
+        inc de ; skip '-'
+        call read_bcd_pair
+        inc de ; skip '-'
+        call read_bcd_pair
+        ; TODO calculate and fill out day-of-week
+        pop hl
+        ret
+
+        ; Read ACSII time and fill out time part of of DATE_STRUCT.
+        ; The input format will be as follows:
+        ;   HH:MM:SS
+        ; Does not check the ':' chars, they can be anything. Does not
+        ; validate number ranges.
+        ; Parameters:
+        ;       HL - Pointer to the date structure, of size DATE_STRUCT_SIZE
+        ;       DE - String source.
+        ; Returns:
+        ;       The time parts of the input date struct have been filled out.
+        ; Modifies: BC
+        PUBLIC ascii_to_time
+ascii_to_time:
+        push hl
+        ld bc, 5
```

Comment:
This should be done in the caller. This routine should not care about the destination in the sense that it is the responsibility of the caller to give a correct destination buffer. So the caller do the `HL + 5`

Reply:
Did this.